pkg/proxy/endpointslicecache_test.go

```diff
@@ -96,43 +96,79 @@ func TestEndpointsMapFromESC(t *testing.T) {
 		"2 slices, overlapping endpoints, some endpoints unready in 1 or both": {
 			namespacedName: types.NamespacedName{Name: "svc1", Namespace: "ns1"},
 			endpointSlices: []*discovery.EndpointSlice{
-				generateEndpointSlice("svc1", "ns1", 1, 10, 3, []string{}, []*int32{utilpointer.Int32Ptr(80)}),
-				generateEndpointSlice("svc1", "ns1", 1, 10, 6, []string{}, []*int32{utilpointer.Int32Ptr(80)}),
+				generateEndpointSlice("svc1", "ns1", 1, 10, 3, 999, []string{}, []*int32{utilpointer.Int32Ptr(80)}),
+				generateEndpointSlice("svc1", "ns1", 1, 10, 6, 999, []string{}, []*int32{utilpointer.Int32Ptr(80)}),
 			},
 			expectedMap: map[ServicePortName][]*BaseEndpointInfo{
 				makeServicePortName("ns1", "svc1", "port-0", v1.ProtocolTCP): {
-					&BaseEndpointInfo{Endpoint: "10.0.1.10:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.1:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.2:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.3:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.4:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.5:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.7:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.8:80"},
-					&BaseEndpointInfo{Endpoint: "10.0.1.9:80"},
+					&BaseEndpointInfo{Endpoint: "10.0.1.10:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.1:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.2:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.3:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.4:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.5:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.6:80", IsLocal: false, Ready: false, Serving: false, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.7:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.8:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+					&BaseEndpointInfo{Endpoint: "10.0.1.9:80", IsLocal: false, Ready: true, Serving: true, Terminating: false},
+				},
```

Comment:
This is relatively hard to understand/parse, a comment to explain what's going on could be helpful. I think same might apply for all of the below test cases unless there's a better way to communicate these changes. We've gone from caring about a list of IP addresses (1 dimension) to 4 additional dimensions.

Reply:
Are the test names not descriptive enough?

Reply:
Maybe it's the `generateEndpointSlice` function that is confusing, I'm not sure. It's just not obvious when you look at this why `10.0.1.6:80` is not ready or serving but the rest are. The test below this feels especially confusing in that regard. (Admittedly I think I'm the one who started this confusing test structure, this just expands it...)

Reply:
yeah using modulo to determine ready state is not ideal, but I didn't want to refactor that in this PR. Happy to refactor this to b ea bit readable in a follow-up PR though